pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -400,10 +401,15 @@ func (c *Controller) onServiceEvent(curr interface{}, event model.Event) error {
 			// only add when it is nodePort gateway service
 			c.nodeSelectorsForServices[svcConv.Hostname] = nodeSelector
 			c.Unlock()
-			c.updateServiceNodePortAddresses(svcConv)
+			needsFullPush = c.updateServiceNodePortAddresses(svcConv)
 		} else {
-			c.extractGatewaysFromService(svcConv)
+			needsFullPush = c.extractGatewaysFromService(svcConv)
 		}
+		if needsFullPush {
+			// networks are different, we need to update all eds endpoints
+			c.xdsUpdater.ConfigUpdate(&model.PushRequest{Full: true})
```

Comment:
can we add a trigger reason? although I am not sur there is one defined that makes sense here, could use "global" or a new one

Reply:
Added a new one

Reply:
I think endpoints sort of makes sense.. not sure how these are used.

Reply:
Its metrics mostly, I don't think we have a log (its pretty noisy). So not a huge deal, but helpful when looking at a spike in CPU, etc and trying to figure out what happened